server.js

```diff
@@ -1,11 +1,148 @@
+    acceptedClockSkewMs: -1, // SAML assertion not yet valid fix
+    logoutUrl: 'https://uwbp-paramedics.us.auth0.com/v2/logout',
+    additionalLogoutParams: {
+      client_id: process.env.AUTH0_CLIENT_ID,
+      federated: '',
+      returnTo: 'http://localhost:4000/logout/callback',
+    },
+  },
+  (profile, done) => {
+    // TODO: This will need to be changed when we switch to ADFS
+    const email =
+      profile[
+        'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'
+      ];
+    db.user
+      .findOne({
+        where: { email: { [Op.like]: email } },
+      })
+      .then((matchingUser) => {
+        // Save the nameId and nameIDFormat for logout
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml = {};
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml.nameID = profile.nameID;
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml.nameIDFormat = profile.nameIDFormat;
+        done(
+          matchingUser ? null : new Error('Email is not registered: ' + email),
+          matchingUser
+        );
+      });
+  }
+);
+
+passport.use(samlStrategy);
+
+/** This saves user.email in the session, under req.session.passport.user,
+ *    which is later used by deserializeUser to find the user object and attach 
+ *    it to req.user. After this operation, req.session.passport.user = user.email */ 
+passport.serializeUser((user, done) => {
+  done(null, user.email);
+});
+
+
+/** This is called with the email that was saved in serializeUser as the
+*     first argument to this function. (i.e. email = user.email). We would then
+*     use the user's email to get the actual user object (using db.user.findOne(...))
+*     Afterwards, the user object is attached to req.user */
+passport.deserializeUser(async (email, done) => {
+  done(
+    null,
+    await db.user.findOne({
+      where: { email: { [Op.like]: email } },
+    })
+  );
+});
+
+const app = express();
+
+app.use(
+  session({
+    secret: SESSION_SECRET,
+    resave: false,
+    saveUninitialized: false,
+    // TODO: Handle login cookie expiring
+    // cookie: { maxAge: 1000 },
+  })
+);
+
+app.use(passport.initialize({}));
+app.use(passport.session({}));
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.get('/login', passport.authenticate('saml'));
+
+// TODO: Redirect to frontend auth landing page
+app.post('/login/callback', passport.authenticate('saml'), (req, res) => {
+  if (!req.user) {
+    res.send('Login unsuccessful');
+  } else {
+    res.send('Login successful!');
+  }
+});
+
+app.get('/logout', (req, res) => {
+  if (req.user == null) {
+    res.send('No logged in user!');
+    return;
+  }
+  samlStrategy.logout(req, function (err, uri) {
+    // Clear user login session on our end
+    res.clearCookie('connect.sid');
+    // Redirect to logout callback URL
+    return res.redirect(uri);
```

Comment:
where does `uri` come from?

Reply:
talked about this with you in call, but i'll leave a comment for clarity